client/src/helper-functions/rpHelpers.js

```diff
@@ -1,10 +1,11 @@
 const helper = {
   compileRelatedProductsDataToProps: (relatedProducts, relatedProductsStyles, starRating) => {
-
+    // console.log(relatedProducts, relatedProductsStyles)
     let allPropsObj = {};
     let relatedProductsCopy = Object.assign(relatedProducts);
     let relatedProductsStylesCopy= Object.assign(relatedProductsStyles);
 
+
     relatedProductsCopy.forEach(item => {
```

Comment:
this function like mine for answers, is filtering data received from API - for performance optimization im wondering if cutting the filtering functions and making ajax request would be more efficient, via time complexity

Reply:
I've been wondering this too recently.... Might eventually make sense just make my own requests for all styles and avoid having to use the helper at all.